server/dataset/dataset.py

```diff
@@ -35,6 +36,44 @@ def __init__(self, data_locator, app_config):
         # parameters set by this data adaptor based on the data.
         self.parameters = {}
 
+    # ------------------------------------------------------------------------
+    # Abstract methods for user annotations - implementations handle persistence
+
+    @abstractmethod
```

Comment:
Base dataset interface now declares the persistence hooks (get/save for annotations + genesets). `CxgDataset` supplies the TileDB implementation; if we ever add a different backend we know which methods to implement.